### ISSUE TITLE
RFE/Bug - Delayed deployment currently not available for off-board artillery

### ISSUE THREAD

**Issue by SimonLandmine** [MegaMek/megamek#2020]
At present in MegaMek, if you designate a unit (artillery) to deploy off-board, it is available from the beginning of the game, regardless of whether you've set a delayed deployment for it, although there is nothing to indicate this.

It is therefore impossible to set off-board artillery to only become available in the "Before Turn X" deployment (which can be useful for some escalating scenarios).

Would it be possible to either ...
a) Indicate that delayed deployment isn't available for units deployed off-board, or
b) Apply the deployment delay to off-board units as well as on-board units.

**Comment by sldfgunslinger2766** [comment]
I can confirm this behavior in Dev 0.47.6. The bot controlled Quad A 4 turret deploys off board as advertised, but is available to fire on Turn 1 even after multiple attempts to set and save the deployment round for round 7.

**Comment by sldfgunslinger2766** [comment]
![deployment](https://user-images.githubusercontent.com/40328392/85343121-4fd42f80-b4a9-11ea-85ef-1c7912368e45.PNG)

**Comment by sldfgunslinger2766** [comment]
![deployment II](https://user-images.githubusercontent.com/40328392/85343237-9590f800-b4a9-11ea-9ef5-9dcbff1263fa.PNG)

**Comment by sixlettervariables** [comment]
From `Entity.java`:
```java
    /**
     * Deploy this offboard entity at the previously specified distance and
     * direction. This should only be invoked by the <code>Server</code> after
     * the board has been selected and all the players are ready to start. The
     * side effects of this methods set the unit's position and facing as
     * appropriate (as well as deploying the unit).
     * <p/>
     * Onboard units (units with an offboard distance of zero and a direction of
     * <code>Entity.NONE</code>) will be unaffected by this method.
     */
    public void deployOffBoard() {
        if (null == game) {
            throw new IllegalStateException(
                    "game not set; possible serialization error");
        }
        // N.B. 17 / 2 = 8, but the middle of 1..17 is 9, so we
        // add a bit (because 17 % 2 == 1 and 16 % 2 == 0).
        switch (offBoardDirection) {
            case NONE:
                break;
            case NORTH:
                setPosition(new Coords((game.getBoard().getWidth() / 2)
                        + (game.getBoard().getWidth() % 2),
                        -getOffBoardDistance()));
                setFacing(3);
                setDeployed(true);
                break;
            case SOUTH:
                setPosition(new Coords((game.getBoard().getWidth() / 2)
                        + (game.getBoard().getWidth() % 2), game.getBoard()
                        .getHeight() + getOffBoardDistance()));
                setFacing(0);
                setDeployed(true);
                break;
            case EAST:
                setPosition(new Coords(game.getBoard().getWidth()
                        + getOffBoardDistance(),
                        (game.getBoard().getHeight() / 2)
                                + (game.getBoard().getHeight() % 2)));
                setFacing(5);
                setDeployed(true);
                break;
            case WEST:
                setPosition(new Coords(-getOffBoardDistance(), (game.getBoard()
                        .getHeight() / 2) + (game.getBoard().getHeight() % 2)));
                setFacing(1);
                setDeployed(true);
                break;
        }
    }
```
They automatically get `setDeployed` assigned.

**Comment by sixlettervariables** [comment]
Fixed by #2027.
